### PR TITLE
chore: a standalone package for the userspace Ed25519 implementation

### PR DESCRIPTION
**Pull request by steveluscher**
# Summary

This is the first module we'll extract from the monolithic legacy library. Here we create a module that encapsulates just the Ed25519 implementation.

### CODE REVIEW

**Comment by steveluscher** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1928** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1928?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1929** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1929?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1930** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1930?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
        * **PR #1931** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1931?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1930?utm_source=stack-comment).

**Review by buffalojoec — COMMENTED** [review]
Lgtm, but I'm curious how we plan to handle the modularized "legacy" packages alongside the shiny new ones.

Specifically, are standalone packages that were created to bust up the legacy Web3.js going to also be considered "legacy"? Or do we support these as standalone installs, either alongside the new Web3.js or on their own.

**Comment by steveluscher** [comment]
Both! They're legacy (don't use them) but you will be able to install them as standalone packages. The primary (and hopefully only) use case of this will be legacy `@solana/wallet-adapter` plugins whose wallets will never migrate away from legacy web3.js but need `Transaction` and `VersionedTransaction` instances to survive.

Eventually we'll move this stuff to a `1.x` branch, leave the latest published `1.x` pacakges on npm, and that's that.

**Review by buffalojoec — DISMISSED** [review]

**Comment by buffalojoec** [comment]
Looks like CI is acting up on this change. Maybe lockfile issue?

**Comment by lorisleiva** [comment]
Should we name these extracted libraries such that it's clear they are part of the legacy library?

Primarily to avoid filling up the `@solana/` namespace with legacy libraries that could one day be used for the new one but also to avoid confusions for users trying to cherry pick their libraries.

I'm thinking `@solana/legacy-ed25519`?

**Comment by steveluscher** [comment]
In my evaluation of the iife bundle, this dropped the size of the library by 316 gzipped bytes, for posterity.

File: packages/legacy/library/src/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+export type {Ed25519Keypair} from '@solana/ed25519';
```
Comment (steveluscher) [inline packages/legacy/library/src/index.ts:1]: Oops. We were supposed to export _only_ the type from the main library.

**Pull request closed without merging**